src/games/cards.ts

```diff
@@ -0,0 +1,520 @@
+  }
+
+  if (mainDeckSmaller) {
+    s.cards.splice(0, maxIndex);
+    shuffledCards.push(...s.cards);
+  } else {
+    d.cards.splice(0, maxIndex);
+    shuffledCards.push(...d.cards);
+  }
+
+  d.cards = shuffledCards;
+  secondDeck.amount--;
+  player.inventory.itemInOffHand = getResizedDeck(deck);
+
+  player.world.playSound(
+    player.location,
+    DECK_SHUFFLE_SOUND,
+    SoundCategory.PLAYERS,
+    1,
+    1,
+  );
+
+  await wait(SHUFFLE_COOLDOWN, 'millis');
+  SHUFFLE_COOLDOWNS.delete(player);
+}
+
+/**
+ * Combines two cards into a deck.
+ * @param card Card in offhand
+ * @param otherCard Card in mainhand
+ * @param player
+ */
+function createDeckFromCards(
+  card: ItemStack,
+  otherCard: ItemStack,
+  player: Player,
+) {
+  let cardID = '';
+  const hiddenCard = HIDDEN_CARD.get(card);
+  if (hiddenCard) {
+    cardID = hiddenCard.cardID;
+  } else {
+    cardID = getCardID(card.itemMeta.customModelData);
+  }
+
+  let otherCardID = '';
+  const hiddenOtherCard = HIDDEN_CARD.get(card);
+  if (hiddenOtherCard) {
+    otherCardID = hiddenOtherCard.cardID;
+  } else {
+    otherCardID = getCardID(otherCard.itemMeta.customModelData);
+  }
+
+  const deck = LOW_DECK.create({ cards: [cardID, otherCardID] });
+  otherCard.amount--;
+  player.inventory.itemInOffHand = deck;
+
+  player.world.playSound(
+    player.location,
+    CARD_FLIP_SOUND,
+    SoundCategory.PLAYERS,
+    1,
+    1,
+  );
+}
+
+// ######################
+// SMALL HELPER FUNCTIONS
+function getDeck(item: ItemStack) {
+  const d = FULL_DECK.get(item) || HALF_DECK.get(item) || LOW_DECK.get(item);
+  return d;
+}
+
+function isDeck(item: ItemStack): boolean {
+  return FULL_DECK.check(item) || HALF_DECK.check(item) || LOW_DECK.check(item);
+}
+
+function isCard(item: ItemStack): boolean {
+  return item.type === Material.PRISMARINE_CRYSTALS && !isDeck(item);
+}
+// ######################
+
+/**
+ * Returns new deck with correct model based on amount of cards in deck
+ * @param deck Deck to be resized
+ */
+function getResizedDeck(deck: ItemStack): ItemStack {
+  const d = getDeck(deck);
+  if (!d) return deck;
+  if (!d.cards) return deck;
+
+  //console.log(d.cards); ( GOOD PLACE TO DEBUG CARDS IN DECK)
+
+  let resizedDeck;
+
+  if (d.cards.length >= 35) {
+    resizedDeck = FULL_DECK.create({ cards: d.cards });
+  } else if (d.cards.length >= 18) {
+    resizedDeck = HALF_DECK.create({ cards: d.cards });
+  } else if (d.cards.length > 1) {
+    resizedDeck = LOW_DECK.create({ cards: d.cards });
+  } else {
+    const cardID = d.cards[0];
+    if (!cardID) return deck;
+    return HIDDEN_CARD.create({ cardID: cardID });
+  }
+
+  resizedDeck.lore = [ChatColor.DARK_GRAY + `${d.cards.length} korttia`];
+  return resizedDeck;
+}
+
+/**
+ * Called ONCE on ANY click in PlayerInteractEvent. Handles card/deck related clicks.
+ * @param player Player who clicked
+ * @param isRightClick True if player rightclicked with either hand
+ */
+function ClickedOnce(
```

Comment:
```suggestion
function clickedOnce(
```
This should use normal function naming.